controllers/metal3labelsync_controller_test.go

```diff
@@ -329,6 +334,537 @@ var _ = Describe("Metal3LabelSync controller", func() {
 			},
 		),
 	)
+	Describe("Test labelsync Reconcile functions", func() {
+
+		type testCaseReconcile struct {
+			prefixSet     map[string]struct{}
+			host          *bmh.BareMetalHost
+			machine       *capi.Machine
+			metal3Machine *capm3.Metal3Machine
+			cluster       *capi.Cluster
+			metal3Cluster *capm3.Metal3Cluster
+			annotation    map[string]string
+			expectError   bool
+			expectRequeue bool
+		}
+		DescribeTable("Test reconcile",
+
+			func(tc testCaseReconcile) {
+
+				objects := []client.Object{}
+				if tc.host != nil {
+					objects = append(objects, tc.host)
+				}
+				if tc.cluster != nil {
+					objects = append(objects, tc.cluster)
+				}
+				if tc.metal3Cluster != nil {
+					tc.metal3Cluster.ObjectMeta.Annotations = tc.annotation
+					objects = append(objects, tc.metal3Cluster)
+				}
+				if tc.machine != nil {
+					tc.machine.Spec.InfrastructureRef = v1.ObjectReference{
+						Name:       metal3machineName,
+						Namespace:  namespaceName,
+						Kind:       "Metal3Machine",
+						APIVersion: capm3.GroupVersion.String(),
+					}
+					tc.machine.Spec.ClusterName = clusterName
+					objects = append(objects, tc.machine)
+				}
+				if tc.metal3Machine != nil {
+					objects = append(objects, tc.metal3Machine)
+				}
+
+				c := fake.NewClientBuilder().WithScheme(setupScheme()).WithObjects(objects...).Build()
+				corev1Client := clientfake.NewSimpleClientset(&v1.Node{ObjectMeta: metav1.ObjectMeta{
+					Name: "testNode",
+				}}).CoreV1()
+				mlsReconcile := &Metal3LabelSyncReconciler{
+					Client:         c,
+					ManagerFactory: baremetal.NewManagerFactory(c),
+					Log:            klogr.New(),
+					CapiClientGetter: func(ctx context.Context, c client.Client, cluster *capi.Cluster) (
+						clientcorev1.CoreV1Interface, error,
+					) {
+						return corev1Client, nil
+					},
+					WatchFilterValue: "",
+				}
+				req := reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      "testHost",
+						Namespace: namespaceName,
+					},
+				}
+				result, err := mlsReconcile.Reconcile(context.TODO(), req)
+
+				if tc.expectError {
+					Expect(err).To(HaveOccurred())
+				} else {
+					Expect(err).NotTo(HaveOccurred())
+				}
+				if tc.expectRequeue {
+					Expect(result.Requeue || result.RequeueAfter > 0).To(BeTrue())
+				} else {
+					Expect(result.Requeue || result.RequeueAfter > 0).To(BeFalse())
+				}
+			},
+			Entry("Baremetal host not found", testCaseReconcile{}),
```

Comment:
no expectations?

Reply:
Here `expectError` is false and `expectRequeue` also false however since above we have `if true else ` they should be handled by the else in any case